reference/api/settings.md

```diff
@@ -120,7 +120,7 @@ Get the settings of an index.
 
 Update the settings of an index.
 
-Passing `null` to an index setting will reset it to its default value.
+Sending a value with a type other than the [one specified](/reference/api/settings.md#body) will result in an [error](/reference/errors/error_codes.md#bad-request).
```

Comment:
Based on the specs, all settings accept `null` except typo tolerance, pagination, and faceting. 

We can add this sentence here or 
- add "Passing `null` to this setting will reset it to its default value" for all settings except typo tolerance, pagination, and faceting
- add "Passing a value with a type other than the one specified will result in an [error](/reference/errors/error_codes.md#bad-request)" for all settings except typo tolerance, pagination, and faceting